### PR TITLE
doc/kconfig.md : Update documentation

### PR DESCRIPTION
**Pull request by akshaim**
### Contribution description

Update documentation to add "Appendix D: A few key aspects while exposing a macro to Kconfig"

### Testing procedure

`make doc`

Doxygen build works fine.

@leandrolanzieri

### CODE REVIEW

File: doc/doxygen/src/kconfig/kconfig.md [hunk 1]
```diff
@@ -375,6 +375,23 @@ menuconfig, the file will not be re-built anymore, and any changes by manually
 editing the source files will have no effect. To go back to manual edition
 a `make clean` has to be issued in the application directory.
 
+## Appendix D: A few key aspects while exposing a macro to Kconfig
```
Comment (leandrolanzieri) [inline doc/doxygen/src/kconfig/kconfig.md:378]:
```suggestion
## Appendix D: A few key aspects while exposing a macro to Kconfig {#kconfig-appendix-d}
```

File: doc/doxygen/src/kconfig/kconfig.md [hunk 2]
```diff
@@ -375,6 +375,23 @@ menuconfig, the file will not be re-built anymore, and any changes by manually
...
+A macro that holds a 0 or 1 gets modelled as a bool. References to this macro
```
Comment (leandrolanzieri) [inline doc/doxygen/src/kconfig/kconfig.md:379]:
```suggestion
A macro that holds a 0 or 1 is modelled in Kconfig as a `bool` symbol. References to this macro
```

**Comment by leandrolanzieri** [comment]
Please squash

**Comment by akshaim** [comment]
> Please squash

Done.

**Pull request merged**